ethers-contract/src/error.rs

```diff
@@ -34,7 +34,15 @@ pub trait ContractRevert: AbiDecode + AbiEncode + Send + Sync {
         if !Self::valid_selector(selector) {
             return None
         }
-        <Self as AbiDecode>::decode(&data[4..]).ok()
+
+        if selector == String::selector() {
+            <Self as AbiDecode>::decode(&data[4..]).ok()
+        } else {
+            // Try with and without a prefix, just in case.
+            <Self as AbiDecode>::decode(data)
+                .or_else(|_| <Self as AbiDecode>::decode(&data[4..]))
+                .ok()
+        }
```

Comment:
I see, for the string case, we expect that `Self` is `String`,
if not we try without prefix and with

Reply:
Yes, the `String` decoder doesn't expect the prefix. I thought everything else did, but some example tests failed without the fallback approach. I don't know why, the tests just timed out waiting for Anvil to start, twice - could be unrelated.